packages/dd-trace/src/log/writer.js

```diff
@@ -23,10 +23,18 @@ function withNoop (fn) {
 }
 
 function unsubscribeAll () {
-  debugChannel.unsubscribe(onDebug)
-  infoChannel.unsubscribe(onInfo)
-  warnChannel.unsubscribe(onWarn)
-  errorChannel.unsubscribe(onError)
+  if (debugChannel.hasSubscribers) {
```

Comment:
Was this caught by any existing test? I think all tests were passing even though manual testing with older versions would fail.

Reply:
Unless we're running tests on the versions of node affected by this bug (ie: `>=14.17.0/15.1.0 <14.19.0/16.14.0/17.1.0`) then I'd say we wouldn't have been able to catch it.

Reply:
What I meant was, if we don't have a test for this, what prevents us from getting a regression? I only found the issue as I was investigating a memory leak locally on older Node versions, but otherwise we would never have caught this until a bug report. Can a test be added?

Reply:
I agree that a test would be useful but we would have to run a whole new version of node on it eh...
cc @iunanua

Reply:
Something like this https://github.com/DataDog/dd-trace-js/pull/2796?